realm/realm-library/src/androidTest/java/io/realm/internal/JNIQueryTest.java

```diff
@@ -192,84 +189,74 @@ public void queryWithWrongDataType() {
-        long columnKey5 = table.getColumnKey("float");
-        long columnKey6 = table.getColumnKey("long");
-        long columnKey7 = table.getColumnKey("string");
-        long[] columnKeys = new long[]{columnKey1, columnKey2, columnKey3, columnKey4, columnKey5, columnKey6, columnKey7};
+        String[] columnKeys = new String[]{"binary", "boolean", "date", "double", "float", "long", "string"};
 
         // Queries the table.
         TableQuery query = table.where();
 
-        // Compares strings in non string columns.
-        for (int i = 0; i < 6; i++) {
-                try { query.equalTo(new long[]{columnKeys[i]}, oneNullTable, "string");    fail(); } catch(IllegalArgumentException ignore) {}
-                try { query.notEqualTo(new long[]{columnKeys[i]}, oneNullTable, "string"); fail(); } catch(IllegalArgumentException ignore) {}
-                try { query.beginsWith(new long[]{columnKeys[i]}, oneNullTable, "string"); fail(); } catch(IllegalArgumentException ignore) {}
-                try { query.endsWith(new long[]{columnKeys[i]}, oneNullTable, "string");   fail(); } catch(IllegalArgumentException ignore) {}
-                try { query.like(new long[]{columnKeys[i]}, oneNullTable, "string");       fail(); } catch(IllegalArgumentException ignore) {}
-                try { query.contains(new long[]{columnKeys[i]}, oneNullTable, "string");   fail(); } catch(IllegalArgumentException ignore) {}
-        }
-
+        Mixed L123 = Mixed.valueOf(123);
+        Mixed L321 = Mixed.valueOf(321);
+        Mixed F123 = Mixed.valueOf(123.5F);
+        Mixed F321 = Mixed.valueOf(321.5F);
+        Mixed D123 = Mixed.valueOf(123.5D);
+        Mixed D321 = Mixed.valueOf(321.5D);
+        Mixed date = Mixed.valueOf(new Date());
 
         // Compares integer in non integer columns.
         for (int i = 0; i <= 6; i++) {
-            if (i != 5) {
-                try { query.equalTo(new long[]{columnKeys[i]}, oneNullTable, 123);            fail(); } catch(IllegalArgumentException ignore) {}
-                try { query.notEqualTo(new long[]{columnKeys[i]}, oneNullTable, 123);         fail(); } catch(IllegalArgumentException ignore) {}
-                try { query.lessThan(new long[]{columnKeys[i]}, oneNullTable, 123);           fail(); } catch(IllegalArgumentException ignore) {}
-                try { query.lessThanOrEqual(new long[]{columnKeys[i]}, oneNullTable, 123);    fail(); } catch(IllegalArgumentException ignore) {}
-                try { query.greaterThan(new long[]{columnKeys[i]}, oneNullTable, 123);        fail(); } catch(IllegalArgumentException ignore) {}
-                try { query.greaterThanOrEqual(new long[]{columnKeys[i]}, oneNullTable, 123); fail(); } catch(IllegalArgumentException ignore) {}
-                try { query.between(new long[]{columnKeys[i]}, 123, 321);                     fail(); } catch(IllegalArgumentException ignore) {}
+            if ((i != 5) && (i != 1) && (i != 3) && (i != 4)) {
```

Comment:
Seems very magically? Would it be smarter to change the dataset, or maybe just make a properly named predicate to guide understanding it.

Reply:
I know.. 🙈 I just updated the tests to adopt the new `TableQuery` logic.